engines/python/setup/djl_python/huggingface.py

```diff
@@ -498,39 +533,81 @@ def register_adapter(inputs: Input):
+                message=f"Failed to register LoRA adapter {adapter_name}")
+
+        if adapter_pin:
+            result = _service.pin_lora(adapter_name)
+            if not result:
+                return Output().error(
+                    "error",
+                    message=f"Failed to pin LoRA adapter {adapter_name}")
+    except (ValueError, RuntimeError) as e:
+        if any(msg in str(e)
+               for msg in ("No free lora slots",
+                           "greater than the number of GPU LoRA slots")):
+            raise MemoryError(str(e))
+        return Output().error(
+            f"Failed to register LoRA adapter {adapter_name}", message=str(e))
+
+    logging.info(
+        f"Registered adapter {adapter_name} from {adapter_path} successfully")
+    return Output(message=f"Adapter {adapter_name} registered")
+
+
+def update_adapter(inputs: Input):
+    """
+    Updates lora adapter with the model.
+    """
+    adapter_name = inputs.get_property("name")
+    adapter_path = inputs.get_property("src")
+    adapter_pin = inputs.get_property("pin") == "true"
+    if adapter_name not in _service.adapter_registry:
+        raise ValueError(f"Adapter {adapter_name} not registered.")
+    _service.adapter_registry[adapter_name] = inputs
 
-        if isinstance(_service.hf_pipeline_unwrapped, Pipeline):
-            _service.hf_pipeline_unwrapped.model = _service.model
-    return Output()
+    if adapter_pin:
+        try:
+            result = _service.pin_lora(adapter_name)
+            if not result:
+                return Output().error(
+                    "error",
+                    message=f"Failed to pin LoRA adapter {adapter_name}")
+        except (ValueError, RuntimeError) as e:
+            if any(msg in str(e)
+                   for msg in ("No free lora slots",
+                               "greater than the number of GPU LoRA slots")):
+                raise MemoryError(str(e))
+            return Output().error(f"Failed to pin LoRA adapter {adapter_name}",
+                                  message=str(e))
+
+    logging.info(f"Updated adapter {adapter_name} successfully")
+    return Output(message=f"Adapter {adapter_name} updated")
 
 
 def unregister_adapter(inputs: Input):
     """
     Unregisters lora adapter from the model.
     """
     adapter_name = inputs.get_property("name")
-    logging.info(f"Unregistering adapter {adapter_name}")
-    #TODO: delete in vllm engine as well
     del _service.adapter_registry[adapter_name]
-    if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
-        _service.model.base_model.delete_adapter(adapter_name)
-    return Output()
+    result = _service.remove_lora(adapter_name)
+    if not result:
+        return Output().error(
+            "error", message=f"Failed to remove LoRA adapter {adapter_name}")
+    logging.info(f"Unregistered adapter {adapter_name} successfully")
+    return Output(message=f"Adapter {adapter_name} unregistered")
```

Comment:
I think we need think about this api a bit. Some questions I have:

- We should probably validate that the adapter_name exists in the adapter registry before attempting to delete it. 
- What happens if service.remove_lora fails? Why would this call fail? In scenarios where this fails, what does that mean for the adapter state within the engine? Is there a chance that this adapter continues to occupy memory?